### ISSUE TITLE
Moving three.js outside of whs build

### ISSUE THREAD

**Issue by sasha240100** [WhitestormJS/whitestorm.js#200]
> This issue happens only on webpack (or other module bundler) and only when using `whs` with `three` dependencies together.

**Reason**

Some time ago I experienced a problem when was developing a website with **webpack + whs + three** combination. I had some bugs that are hard to debug.

**Explanation of problem that happens**

WHS points to it build file `build/whitestorm.js` which is a UMD implementation of whs and three together. And when you also import `three` like that:

```javascript
import * as WHS from 'whs';
import * as THREE from 'three';
```

It will result into a following order of bundled files:

#### `bundle.js`

- **whs** [UMD]
  - three.js _<- 1_
  - whs wrapper (components, modules, ...)
- **three.js** _<- 2_
- other modules

See that `three.js` dependency is bundled twice. **It's incorrect**.

**Solution**

The solution is simple - move `three.js` to whs externals. So **instead of**:

```html
<script src="whitestorm.js"></script> <!-- ~600Kb -->
```

**You'll have:**
```html
<script src="three.js"></script> <!-- ~500Kb -->
<script src="whitestorm.js"></script> <!-- ~100Kb -->
```

###### Version:
 - [x] v2.x.x
 - [x] v1.x.x

###### Issue type:
 - [x] Bug
 - [x] Proposal/Enhancement
 - [ ] Question

------

<details> 
<summary> <b>Tested on: </b> </summary>

###### Desktop
   - [ ] Chrome
   - [ ] Chrome Canary
   - [ ] Chrome dev-channel
   - [ ] Firefox
   - [ ] Opera
   - [ ] Microsoft IE
   - [ ] Microsoft Edge

###### Android
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Opera

###### IOS
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Opera

</details>

**Comment by t4r0** [comment]
@sasha240100 but isn't this the problem why we ported three.js into whs?

**Comment by sasha240100** [comment]
@t4r0 I ported `three.js` into `whs` because it requires specific version of `three`. But know such problem we face. It seems @thejmazz has this problem too. It results in larger bundle (2x three.js) and bugs in Three.js

**Comment by sasha240100** [comment]
@t4r0 So it is like a regression from my decision. We need to fix it and make it as it should be

**Comment by t4r0** [comment]
What I am thinking is that perhaps this is not an issue related to whs itself, it is related to your settings when bundling whs with three. My question is why are you using three aside whs? Wouldn't it be better to expose the Three module inside whs, somehting like.

`import {THREE} from 'whs/threejs';`

**Comment by sasha240100** [comment]
@t4r0 This is an issue of whs webpack configuration. Three.js should be moved to `externals`.

**Comment by t4r0** [comment]
alright I get it

**Comment by sasha240100** [comment]
Also, it is a possible fix of #198

**Comment by thejmazz** [comment]
Confirmed, `import * as THREE from 'whs/src/three'` is a fix of #198.

**Comment by sasha240100** [comment]
@thejmazz See #200. You should be able to use `impprt * as THREE from 'three'` in next version or if you will use it latest commit in beta and do `npm link`
